src/blob_tree/mod.rs

```diff
@@ -440,33 +431,41 @@ impl AbstractTree for BlobTree {
         self.index.sealed_memtable_count()
     }
 
-    fn flush_memtable(
+    fn get_flush_lock(&self) -> MutexGuard<'_, ()> {
+        self.index.get_flush_lock()
+    }
+
+    fn flush_to_tables(
         &self,
-        table_id: TableId,
-        memtable: &Arc<Memtable>,
-        eviction_seqno: SeqNo,
-    ) -> crate::Result<Option<(Table, Option<BlobFile>)>> {
-        use crate::{file::TABLES_FOLDER, table::Writer as TableWriter};
+        stream: impl Iterator<Item = crate::Result<InternalValue>>,
+    ) -> crate::Result<Option<(Vec<Table>, Option<Vec<BlobFile>>)>> {
+        use crate::{coding::Encode, file::TABLES_FOLDER, table::multi_writer::MultiWriter};
 
-        let table_folder = self.index.config.path.join(TABLES_FOLDER);
+        let start = std::time::Instant::now();
 
-        log::debug!("Flushing memtable & performing key-value separation");
-        log::debug!("=> to table in {}", table_folder.display());
-        log::debug!("=> to blob file at {}", self.blobs_folder.display());
+        let table_folder = self.index.config.path.join(TABLES_FOLDER);
 
-        let mut table_writer =
-            TableWriter::new(table_folder.join(table_id.to_string()), table_id, 0)?
-                // TODO: apply other policies
-                .use_data_block_compression(self.index.config.data_block_compression_policy.get(0))
-                .use_bloom_policy({
-                    use crate::config::FilterPolicyEntry::{Bloom, None};
-                    use crate::table::filter::BloomConstructionPolicy;
+        log::debug!("Flushing memtable(s) & performing key-value separation");
+        log::debug!("=> to table(s) in {}", table_folder.display());
+        log::debug!("=> to blob file(s) at {}", self.blobs_folder.display());
 
-                    match self.index.config.filter_policy.get(0) {
-                        Bloom(policy) => policy,
-                        None => BloomConstructionPolicy::BitsPerKey(0.0),
-                    }
-                });
+        let mut table_writer = MultiWriter::new(
+            table_folder.clone(),
+            self.index.table_id_counter.clone(),
+            64_000_000,
+            0,
+        )?
+        // TODO: 3.0.0 apply other policies
```

Comment:
I assume this will include:

.use_data_block_restart_interval(data_block_restart_interval)
.use_index_block_restart_interval(index_block_restart_interval)
.use_data_block_size(data_block_size)
.use_data_block_hash_ratio(data_block_hash_ratio)